<?php

namespace OpenExam\Models;

use Exception;
use OpenExam\Tests\Phalcon\TestModel;
use stdClass;

/**
 * Generated by PHPUnit_SkeletonGenerator 1.2.1 on 2014-09-15 at 19:03:31.
 * 
 * @author Anders LÃ¶vgren (Computing Department at BMC, Uppsala University)
 */
class ExamTest extends TestModel
{

        public function __construct()
        {
                parent::__construct(new Exam());
        }

        /**
         * @covers OpenExam\Models\Exam::properties
         * @group model
         */
        public function testProperties()
        {
                $values = array(
                        'creator' => 'user1',
                        'orgunit' => 'orgunit1',
                        'grades'  => new stdClass(array('key1' => 'val1')),
                        'name'    => 'name1'
                );

                try {
                        $helper = new TestModel(new Exam());
                        $helper->tryPersist();
                        self::fail("Excepted constraint violation exception");
                } catch (Exception $exception) {
                        // Expected exception
                }

                try {
                        $helper = new TestModel(new Exam());
                        $helper->tryPersist($values);
                } catch (Exception $exception) {
                        self::fail($exception);
                }

                $helper->checkDefaults(array(
                        'details'  => 3,
                        'decoded'  => false,
                        'testcase' => false,
                        'lockdown' => false
                ));

                $values = array(
                        'updated' => date('Y-m-d H:i:s'),
                        'created' => date('Y-m-d H:i:s'),
                        'details' => 3,
                        'creator' => 'user1',
                        'orgunit' => 'orgunit1',
                        'grades'  => new stdClass(array('key1' => 'val1')),
                        'name'    => 'name1'
                );
                try {
                        $helper = new TestModel(new Exam());
                        $helper->tryPersist($values);
                } catch (Exception $exception) {
                        self::fail($exception);
                }

                $values = array(
                        'updated'   => date('Y-m-d H:i:s'),
                        'created'   => date('Y-m-d H:i:s'),
                        'details'   => 3,
                        'name'      => 'Name1',
                        'descr'     => 'Description1',
                        'starttime' => date('Y-m-d H:i:s'),
                        'endtime'   => date('Y-m-d H:i:s'),
                        'creator'   => 'user1',
                        'decoded'   => true,
                        'orgunit'   => 'Orgunit1',
                        'grades'    => new stdClass(),
                        'testcase'  => true,
                        'lockdown'  => true
                );
                try {
                        $helper = new TestModel(new Exam());
                        $helper->tryPersist($values);
                } catch (Exception $exception) {
                        self::fail($exception);
                }
        }

        /**
         * @covers OpenExam\Models\Exam::getSource
         * @group model
         */
        public function testGetSource()
        {
                $expect = "exams";
                $actual = $this->object->getSource();
                self::assertNotNull($actual);
                self::assertEquals($expect, $actual);
        }

}
